tests/core/test_library.py

```diff
@@ -437,52 +444,182 @@ def setUp(self):  # noqa: N802
 
         self.core = core.Core(mixer=None, backends=[self.backend])
 
-    def test_browse_backend_get_root_exception_gets_ignored(self, logger):
+
+@mock.patch('mopidy.core.library.logger')
+class BrowseBadBackendTest(MockBackendCoreLibraryBase):
+
+    def test_backend_raises_exception_for_root(self, logger):
         # Might happen if root_directory is a property for some weird reason.
         self.library.root_directory.get.side_effect = Exception
         self.assertEqual([], self.core.library.browse(None))
         logger.exception.assert_called_with(mock.ANY, 'DummyBackend')
 
-    def test_browse_backend_browse_uri_exception_gets_ignored(self, logger):
+    def test_backend_returns_none_for_root(self, logger):
+        self.library.root_directory.get.return_value = None
+        self.assertEqual([], self.core.library.browse(None))
+        logger.error.assert_called_with(mock.ANY, 'DummyBackend', mock.ANY)
```

Comment:
I think we should use pytest's `caplog` fixture instead: https://bitbucket.org/memedough/pytest-capturelog/src/039cb3be01c9c6480d3907c47247f5bb0744eb36/README?at=default